From: Petter Reinholdtsen <pere@hungry.com>
Date: Sun, 14 Jul 2024 10:20:09 +0200
Subject: calibre: fix appstream ID warning (Closes: #1076311)

Package: calibre
Version: 6.13.0+repack-2
Tags: patch

<URL: https://appstream.debian.org/sid/main/issues/calibre.html >
contain the following warning about the content of the appstream
metadata file:

  * asv-cid-desktopapp-is-not-rdns
    calibre-gui.metainfo.xml:3 - calibre-gui.desktop
    The component ID is not a reverse domain-name. Please update the ID
    to avoid future issues and be compatible with all AppStream
    implementations. You may also consider to update the name of the
    accompanying .desktop file to follow the latest version of the
    Desktop-Entry specification and use a rDNS name for it as well. In
    any case, do not forget to mention the new desktop-entry in a
    <launchable/> tag for this component to keep the application
    launchable from software centers and the .desktop file data
    associated with the metainfo data.

I believe the following untested patch will solve the issue.  Sadly I
lack the required free disk space available to test this patch.

Signed-off-by: YOKOTA Hiroshi <yokota.hgml@gmail.com>
---
 src/calibre/linux.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/calibre/linux.py b/src/calibre/linux.py
index daf8e6d..ec30e76 100644
--- a/src/calibre/linux.py
+++ b/src/calibre/linux.py
@@ -1242,6 +1242,7 @@ def get_appdata():
                 ),
             ),
             'desktop-id': 'calibre-gui.desktop',
+            'appstream-id': 'com.calibre_ebook.calibre',
             'include-releases': True,
         },
         'calibre-ebook-edit': {
@@ -1271,6 +1272,7 @@ def get_appdata():
                 ),
             ),
             'desktop-id': 'calibre-ebook-edit.desktop',
+            'appstream-id': 'com.calibre_ebook.calibre.ebook_edit',
         },
         'calibre-ebook-viewer': {
             'name': 'calibre - E-book Viewer',
@@ -1292,6 +1294,7 @@ def get_appdata():
                 ),
             ),
             'desktop-id': 'calibre-ebook-viewer.desktop',
+            'appstream-id': 'com.calibre_ebook.calibre.ebook_viewer',
         },
     }
 
@@ -1365,7 +1368,7 @@ def write_appdata(key, entry, base, translators):
                 description.append(E.p(tp, **{'{http://www.w3.org/XML/1998/namespace}lang': lang}))
 
     root = E.component(
-        E.id(key + '.desktop'),
+        E.id(entry['appstream-id']),
         E.name(entry['name']),
         E.metadata_license('CC0-1.0'),
         E.project_license('GPL-3.0'),
